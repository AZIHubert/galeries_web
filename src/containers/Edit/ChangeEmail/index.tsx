import { useFormik } from 'formik';
import * as React from 'react';
import {
  useSelector,
} from 'react-redux';

import { changeEmailSchema } from '#helpers/schemas';
import { loadingSelector } from '#store/selectors';
import Field from '#components/Field';

const initialValues = {
  password: '',
};

const ChangeEmail = () => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {},
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: changeEmailSchema,
  });
  const loading = useSelector(loadingSelector);
  return (
    <div>
      <p>
        change your email
      </p>
      <p>
        Enter your password to receive a message on your current adress.
        This message contain a link to update your email.
      </p>
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          error={formik.errors.password}
          fieldTestId='field'
          id='password'
          label='password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          touched={formik.touched.password}
          type='password'
          value={formik.values.password}
        />
        <button
          disabled={loading}
          type='submit'
        >
          {loading ? 'loading' : 'change your email'}
        </button>
      </form>
    </div>
  );
};

export default ChangeEmail;
