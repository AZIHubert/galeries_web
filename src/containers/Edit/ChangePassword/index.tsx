import * as React from 'react';
import { useFormik } from 'formik';
import {
  useSelector,
} from 'react-redux';

import Field from '#components/Field';
import { changePasswordSchema } from '#helpers/schemas';

import { loadingSelector } from '#store/selectors';

const initialValues = {
  currentPassword: '',
  newPassword: '',
  confirmNewPassword: '',
};

const ChangePassword = () => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {},
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: changePasswordSchema,
  });
  const loading = useSelector(loadingSelector);
  return (
    <div>
      <p>
        change password
      </p>
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          error={formik.errors.currentPassword}
          fieldTestId='currentPassword'
          id='currentPassword'
          label='current password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          touched={formik.touched.currentPassword}
          type='password'
          value={formik.values.currentPassword}
        />
        <Field
          disabled={loading}
          error={formik.errors.newPassword}
          fieldTestId='newPassword'
          id='newPassword'
          label='new password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          touched={formik.touched.newPassword}
          type='password'
          value={formik.values.newPassword}
        />
        <Field
          disabled={loading}
          error={formik.errors.confirmNewPassword}
          fieldTestId='confirmNewPassword'
          id='confirmNewPassword'
          label='confirm new password'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          touched={formik.touched.confirmNewPassword}
          type='password'
          value={formik.values.confirmNewPassword}
        />
        <button
          disabled={loading}
          data-testid='submitButton'
          type='submit'
        >
          {loading ? 'loading' : 'change your password'}
        </button>
      </form>
    </div>
  );
};

export default ChangePassword;
