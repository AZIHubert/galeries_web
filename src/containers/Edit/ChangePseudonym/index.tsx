import { useFormik } from 'formik';
import * as React from 'react';
import {
  useSelector,
} from 'react-redux';

import Button from '#components/Button';
import Field from '#components/Field';
import RequiredField from '#components/RequiredField';

import { pseudonymSchema } from '#helpers/schemas';

import {
  loadingSelector,
} from '#store/selectors';

const initialValues: form.PseudonymI = {
  pseudonym: '',
};

const ChangePeudonym = () => {
  const formik = useFormik({
    initialValues,
    onSubmit: () => {},
    validateOnBlur: true,
    validateOnChange: false,
    validationSchema: pseudonymSchema,
  });
  const loading = useSelector(loadingSelector);

  return (
    <div>
      <form onSubmit={formik.handleSubmit}>
        <Field
          disabled={loading}
          error={
            formik.errors.pseudonym
          }
          fieldTestId='field'
          id='pseudonym'
          label='pseudonym'
          onBlur={formik.handleBlur}
          onChange={formik.handleChange}
          required
          styles={{
            marginBottom: 25,
          }}
          touched={formik.touched.pseudonym}
          type='text'
          value={formik.values.pseudonym}
        />
        <RequiredField />
        <Button.Gradiant
          disabled={loading}
          styles={{
            marginTop: 20,
          }}
          type='submit'
          title='change your pseudonym'
        />
      </form>
    </div>
  );
};

export default ChangePeudonym;
