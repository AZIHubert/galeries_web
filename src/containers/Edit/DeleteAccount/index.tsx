import * as React from 'react';
import { useSelector } from 'react-redux';

import Modal from '#components/Modal';

import { loadingSelector } from '#store/selectors';

import ModalDelete from './ModalDelete';

const DeleteAccount = () => {
  const loading = useSelector(loadingSelector);
  const [openModal, setOpenModal] = React.useState<boolean>(false);

  const handleClose = () => {
    if (!loading) {
      setOpenModal(false);
    }
  };

  return (
    <div>
      <p>
        delete your account
      </p>
      <p>
        Once you delete your account, there is no going back. Please be certain.
      </p>
      <button
        onClick={() => setOpenModal((nextState) => !nextState)}
      />
      <Modal.Portal
        modalTestId='modal'
        open={openModal}
        handleClose={handleClose}
      >
        <ModalDelete />
      </Modal.Portal>
    </div>
  );
};

export default DeleteAccount;
