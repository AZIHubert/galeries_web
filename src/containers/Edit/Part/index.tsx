import * as React from 'react';

import Text from '#components/Text';

import {
  ChildrenContainer,
  Container,
  TitleContainer,
} from './styles';

interface PartI {
  danger?: boolean;
  title?: string;
}

const Part: React.FC<PartI> = ({
  children,
  danger = false,
  title,
}) => (
  <Container>
    { title ? (
      <TitleContainer
        danger={danger}
      >
        <Text
          color={danger ? 'danger' : 'black'}
          styles={{
            fontSize: 1.3,
            marginBottom: 12,
          }}
        >
          {title}
        </Text>
      </TitleContainer>
    ) : null}
    <ChildrenContainer>
      { children }
    </ChildrenContainer>
  </Container>
);

export default Part;
