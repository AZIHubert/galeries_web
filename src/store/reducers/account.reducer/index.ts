import { ACCOUNT_SET } from '#store/actions';

interface InitialStateI {
  errors: form.AccountI,
  status: store.Status,
}

const initialState: InitialStateI = {
  errors: {
    deleteAccountSentence: '',
    password: '',
    userNameOrEmail: '',
  },
  status: 'pending',
};

export default (
  state = initialState,
  action: store.ActionI,
) => {
  switch (action.type) {
    case ACCOUNT_SET:
      return {
        ...state,
        ...action.payload ? action.payload.data : undefined,
        errors: {
          ...state.errors,
          ...action.payload ? action.payload.data.errors : undefined,
        },
      };
    default:
      return state;
  }
};
