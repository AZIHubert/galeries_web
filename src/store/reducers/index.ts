import { combineReducers } from 'redux';

import login from './login.reducer'; // loginError
import notification from './notification.reducer';
import sendConfirmation from './sendConfirmation.reducer'; // sendConfirmationError
import sendResetPassword from './sendResetPassword.reducer'; // sendResetPasswordError
import signin from './signin.reducer'; // SigninError
import user from './user.reducer';
import ui from './ui.reducer';

export default combineReducers({
  login,
  notification,
  sendConfirmation,
  sendResetPassword,
  signin,
  ui,
  user,
});
