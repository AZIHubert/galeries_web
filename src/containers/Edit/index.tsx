import * as React from 'react';
import {
  useDispatch,
  useSelector,
} from 'react-redux';

import Footer from '#containers/Footer';

import { resetUpdatePassword } from '#store/actions';
import { userSelector } from '#store/selectors';

import ChangeEmail from './ChangeEmail';
import ChangePassword from './ChangePassword';
import DeleteAccount from './DeleteAccount';
import Part from './Part';

import {
  Container,
} from './styles';

const Edit = () => {
  const dispatch = useDispatch();
  const user = useSelector(userSelector);

  React.useEffect(() => () => {
    dispatch(resetUpdatePassword());
  }, []);

  const isNotAFacebookOrGoogleUser = user && !user.facebookId && !user.googleId;

  return (
    <Container>
      {isNotAFacebookOrGoogleUser && (
        <>
          <Part
            title='Change your password'
          >
            <ChangePassword />
          </Part>
          <Part
            title='Change your email'
          >
            <ChangeEmail />
          </Part>
          <Part
            danger
            title='Delete your account'
          >
            <DeleteAccount />
          </Part>
        </>
      )}
      <Footer />
    </Container>
  );
};

export default Edit;
