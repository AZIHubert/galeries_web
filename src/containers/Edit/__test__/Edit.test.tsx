import { cleanup } from '@testing-library/react';
import * as React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';

import ThemeProvider from '#contexts/ThemeContext';
import reducers from '#store/reducers';

import Edit from '../index';

const mockedStore = createStore(reducers);

const Container = () => (
  <Provider store={mockedStore}>
    <ThemeProvider>
      <Edit />
    </ThemeProvider>
  </Provider>
);

describe('Edit', () => {
  afterEach(cleanup);
  it('renders without crashing', () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
