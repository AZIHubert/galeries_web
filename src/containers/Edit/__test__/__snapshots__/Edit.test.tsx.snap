// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit renders without crashing 1`] = `
Array [
  <div>
    <p>
      change password
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-bdfBwQ bHhtWb"
      >
        <label
          className="sc-hKgILt bXTMsJ"
          htmlFor="currentPassword"
        >
          <p>
            current password
          </p>
        </label>
        <input
          className="sc-dlfnbm grJsJX"
          data-testid="currentPassword"
          disabled={false}
          id="currentPassword"
          name="currentPassword"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="sc-bdfBwQ bHhtWb"
      >
        <label
          className="sc-hKgILt bXTMsJ"
          htmlFor="newPassword"
        >
          <p>
            new password
          </p>
        </label>
        <input
          className="sc-dlfnbm grJsJX"
          data-testid="newPassword"
          disabled={false}
          id="newPassword"
          name="newPassword"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="sc-bdfBwQ bHhtWb"
      >
        <label
          className="sc-hKgILt bXTMsJ"
          htmlFor="confirmNewPassword"
        >
          <p>
            confirm new password
          </p>
        </label>
        <input
          className="sc-dlfnbm grJsJX"
          data-testid="confirmNewPassword"
          disabled={false}
          id="confirmNewPassword"
          name="confirmNewPassword"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <button
        disabled={false}
        type="submit"
      >
        change your password
      </button>
    </form>
  </div>,
  <div>
    <p>
      change your email
    </p>
    <p>
      Enter your password to receive a message on your current adress. This message contain a link to update your email.
    </p>
    <form
      onSubmit={[Function]}
    >
      <div
        className="sc-bdfBwQ bHhtWb"
      >
        <label
          className="sc-hKgILt bXTMsJ"
          htmlFor="password"
        >
          <p>
            password
          </p>
        </label>
        <input
          className="sc-dlfnbm grJsJX"
          data-testid="field"
          disabled={false}
          id="password"
          name="password"
          onBlur={[Function]}
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <button
        disabled={false}
        type="submit"
      >
        change your email
      </button>
    </form>
  </div>,
  <div>
    <p>
      delete your account
    </p>
    <p>
      Once you delete your account, there is no going back. Please be certain.
    </p>
    <button
      onClick={[Function]}
    />
  </div>,
]
`;
